snapcraft/yaml.py

```diff
@@ -38,14 +38,26 @@
 
 @jsonschema.FormatChecker.cls_checks('file-path')
 def _validate_file_exists(instance):
-    return os.path.exists(instance)
+    if not os.path.exists(instance):
+        raise jsonschema.exceptions.ValidationError(
+            'Specified file \'{}\' does not exist'.format(instance))
+
+    return True
 
 
 @jsonschema.FormatChecker.cls_checks('icon-path')
 def _validate_icon(instance):
     normalized_icon = instance.lower()
-    ext = normalized_icon.endswith('.png') or normalized_icon.endswith('.svg')
-    return os.path.exists(instance) and ext
+    if not (normalized_icon.endswith('.png') or
```

Comment:
You could use os.path.splitext once, and then just check for equality.
I'm not too convinced that it would be much clearer, so feel free to ignore.